warehouse/static/js/warehouse/utils/messages-access.js

```diff
@@ -0,0 +1,109 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// the value for 'messagesAccessLocaleData' is set by webpack.plugin.localize.js
+// default is en
+var messagesAccessLocaleData = {"": {"language": "en", "plural-forms": "nplurals = 2; plural = (n != 1)"}};
+
+// the value for 'messagesAccessPluralFormFunction' is set by webpack.plugin.localize.js
+// default is en
+var messagesAccessPluralFormFunction = function (n) {
+  let nplurals, plural;
+  nplurals = 2; plural = (n != 1);
+  return {total: nplurals, index: ((nplurals > 1 && plural === true) ? 1 : (plural ? plural : 0))};
+};
+
+/**
+ * Get the translation using num to choose the appropriate string.
+ *
+ * When importing this function, it must be named in a particular way to be recognised by babel
+ * and have the translation strings extracted correctly.
+ * Function 'ngettext' for plural extraction.
+ *
+ * Any placeholders must be specified as %1, %2, etc.
+ *
+ * @example
+ * import { ngettext } from "warehouse/utils/messages-access";
+ * // For a singular and plural and placeholder string:
+ * ngettext("About a minute ago", "About %1 minutes ago", numMinutes, numMinutes);
+
+ * @param singular {string} The default string for the singular translation.
+ * @param plural {string} The default string for the plural translation.
+ * @param num {number} The number to use to select the appropriate translation.
+ * @param extras {string} Additional values to put in placeholders.
+ * @returns {string} The translated text.
+ * @see https://github.com/guillaumepotier/gettext.js
+ * @see https://www.gnu.org/software/gettext/manual/gettext.html#Language-specific-options
+ * @see https://docs.pylonsproject.org/projects/pyramid/en/latest/api/i18n.html#pyramid.i18n.Localizer.pluralize
+ */
+export function ngettext(singular, plural, num, ...extras) {
+  return ngettextCustom(singular, plural, num, extras, messagesAccessLocaleData, messagesAccessPluralFormFunction);
+}
+
+/**
+ * Get the singular translation.
+ *
+ * When importing this function, it must be named in a particular way to be recognised by babel
+ * and have the translation strings extracted correctly.
+ * Function 'gettext' for singular extraction.
+ *
+ * Any placeholders must be specified as %1, %2, etc.
+ *
+ * @example
+ * import { gettext } from "warehouse/utils/messages-access";
+ * // For a singular only string:
+ * gettext("Just now");
+ *
+ * @param singular {string} The default string for the singular translation.
+ * @param extras {string} Additional values to put in placeholders.
+ * @returns {string} The translated text.
+ */
+export function gettext(singular, ...extras) {
+  return ngettext(singular, null, 1, ...extras);
+}
+
+export function ngettextCustom(singular, plural, num, extras, data, pluralForms) {
+  // this function allows for testing
```

Comment:
question: is this comment still accurate, if `ngettext` is basically a wrapper for this, with some defaults?

Reply:
addressed in d32b4472ebf326109af61e0cce90e2f7015589c0